joint_limits/test/test_simple_joint_limiter.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEST_SIMPLE_JOINT_LIMITER_HPP_
+#define TEST_SIMPLE_JOINT_LIMITER_HPP_
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "joint_limits/joint_limiter_interface.hpp"
+#include "joint_limits/joint_limits.hpp"
+#include "pluginlib/class_loader.hpp"
+#include "rclcpp/duration.hpp"
+#include "rclcpp/node.hpp"
+#include "trajectory_msgs/msg/joint_trajectory_point.hpp"
+
+const double COMMON_THRESHOLD = 0.0011;
+
+#define CHECK_STATE_SINGLE_JOINT(tested_traj_point, idx, expected_pos, expected_vel, expected_acc) \
+  EXPECT_NEAR(tested_traj_point.positions[idx], expected_pos, COMMON_THRESHOLD);                   \
+  EXPECT_NEAR(tested_traj_point.velocities[idx], expected_vel, COMMON_THRESHOLD);                  \
+  EXPECT_NEAR(tested_traj_point.accelerations[idx], expected_acc, COMMON_THRESHOLD);
+
+#define INTEGRATE(cur, des, dt)                                                      \
+  cur.positions[0] += des.velocities[0] * dt + 0.5 * des.accelerations[0] * dt * dt; \
+  cur.velocities[0] += des.accelerations[0] * dt;
+
+using JointLimiter = joint_limits::JointLimiterInterface<joint_limits::JointLimits>;
+
+class SimpleJointLimiterTest : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    auto testname = ::testing::UnitTest::GetInstance()->current_test_info()->name();
```

Comment:
This is cool!

Reply:
This has been found in another test in another package (cannot quickly find where) or in stackoverflow (like here https://stackoverflow.com/questions/55650559/how-to-extract-test-name-in-google-tests-case-method) because I needed the name to associate the correct params to that node (first used it in low_pass_filter). 

How would you "acknowledge" the source material ? I think for very short snippets of stackoverflow unless it is an algorithm someone came up with, it is accepted to just copy-paste. I would love to have every person be "thanked" for the snippets, but I know the code would become stuffed with comments about where the snippets come from.
What is the good practice ?

Reply:
This is a line of code. You can vote up and this is OK in this case.